### ISSUE TITLE
_includes/seo.htm is missing post.excerpt from seo_description 

### ISSUE THREAD

**Issue by funvill** [mmistakes/minimal-mistakes#2964]
## Environment

- Minimal Mistakes version: 4.23.0 
- Git repository URL: https://github.com/funvill/funvill.github.io 
- GitHub Pages hosted (if yes provide URL to site): https://blog.abluestar.com/

## Expected behavior

When linking to a "post" in a system that supports Open Graph such as Slack or Twitter. The site description is used instead of the first paragraph of the "post" as expected. 

The first paragraph is used as an Open Graph description (`og:description`) for pages and collections as expected. 

I expected the default Open Graph description of a post to be the first paragraph of the post instead of the site description. 

## Fix. 

[Line 23 of _includes/seo.htm](https://github.com/mmistakes/minimal-mistakes/blob/e134d13e457a71900f1d5732cb8668205627ba66/_includes/seo.html#L23)  needs to be updated

From 

`{%- assign seo_description = page.description | default: page.excerpt | default: site.description -%}`

to 

`{%- assign seo_description = page.description | default: page.excerpt | default: post.excerpt | default: site.description -%}`

**Comment by funvill** [comment]
as @iBug pointed out in this pull request https://github.com/mmistakes/minimal-mistakes/pull/2965 we can not rely on `post.excerpt` as it's not an "automatic variable in Jekyll".

Upon more research into this issue using the demo https://mmistakes.github.io/minimal-mistakes/ theme, it doesn't appear to be an issue on the demo site. So it must be unique to my installation. I am going to close this issue do more research and open a new issue if needed. 

Thanks for your help @iBug

**Comment by mmistakes** [comment]
Can't reproduce this on the demo site, so maybe there's something specific going on in your repo.

Example post on the demo site: https://mmistakes.github.io/minimal-mistakes/jekyll/welcome-to-jekyll/
And looking at the sourcecode I see the following, which is taking the first paragraph of text and using that as `page.excerpt` by Jekyll.

```html
<meta name="description" content="You’ll find this post in your _posts directory. Go ahead and edit it and re-build the site to see your changes. You can rebuild the site in many different ways, but the most common way is to run jekyll serve, which launches a web server and auto-regenerates your site when a file is updated.">
```

And on a post with a custom excerpt it is always working as expected.

https://mmistakes.github.io/minimal-mistakes/layout/uncategorized/layout-excerpt-defined/

```html
<meta name="description" content="This is a user-defined post excerpt. It should be displayed in place of the post content in archive-index pages.">
```

**Comment by funvill** [comment]
I am sorry @mmistakes this issue was badly worded, didn't provide good examples, and incorrectly described the problem. It looks like I closed this issue too late and you already were in the process of responding. Sorry for wasting your time on this badly worded issue. 

I have created a new issue using the demo site for examples with hopefully better descriptions and examples. https://github.com/mmistakes/minimal-mistakes/issues/2966
